libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SetProperties.cs

```diff
@@ -81,6 +81,9 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 dc.State.SetValue(propValue.Property.GetValue(dc.State), value);
             }
 
+            // Explicit state change resets retries
+            dc.State.RemoveValue(DialogPath.Retries);
```

Comment:
This reset in retries happens regardless of having a change in state. Should we add a flag inside the 'foreach' to indicate this change needs to be made after the loop?

Reply:
Just calling this is enough to reset even if nothing changed about the state. Retries is used to identify when the same conceptual query is asked multiple times.  Any memory setting indicates a response was processed and we should reset it.

---
In reply to: [490538953](https://github.com/microsoft/botbuilder-dotnet/pull/4665#discussion_r490538953) [](ancestors = 490538953)